diff --git a/components/cxx/cxx_exception_stubs.cpp b/components/cxx/cxx_exception_stubs.cpp
index 7e794755c07..f3de729347b 100644
--- a/components/cxx/cxx_exception_stubs.cpp
+++ b/components/cxx/cxx_exception_stubs.cpp
@@ -6,7 +6,10 @@
 #include <cstdlib>
 #include <cstdio>
 #include <exception>
+#if defined(_LIBCPP_VERSION)
+#else
 #include <bits/functexcept.h>
+#endif
 #include <sdkconfig.h>
 
 #ifndef CONFIG_COMPILER_CXX_EXCEPTIONS
diff --git a/components/cxx/cxx_guards.cpp b/components/cxx/cxx_guards.cpp
index e4f9f905a66..98a6feaceda 100644
--- a/components/cxx/cxx_guards.cpp
+++ b/components/cxx/cxx_guards.cpp
@@ -15,7 +15,11 @@
 #include "freertos/semphr.h"
 #include "freertos/task.h"
 
+#if defined(_LIBCPP_VERSION)
+typedef uint64_t __guard;
+#else
 using __cxxabiv1::__guard;
+#endif
 
 static SemaphoreHandle_t s_static_init_mutex = NULL;        //!< lock used for the critical section
 static SemaphoreHandle_t s_static_init_wait_sem = NULL;     //!< counting semaphore used by the waiting tasks
@@ -26,8 +30,13 @@ static size_t s_static_init_max_waiting_count = 0;          //!< maximum ever va
 #endif
 
 extern "C" int __cxa_guard_acquire(__guard* pg);
+#if defined(_LIBCPP_VERSION)
+extern "C" void __cxa_guard_release(__guard* pg);
+extern "C" void __cxa_guard_abort(__guard* pg);
+#else
 extern "C" void __cxa_guard_release(__guard* pg) throw();
 extern "C" void __cxa_guard_abort(__guard* pg) throw();
+#endif
 extern "C" void __cxa_guard_dummy(void);
 
 /**
@@ -167,7 +176,11 @@ extern "C" int __cxa_guard_acquire(__guard* pg)
     return ret;
 }
 
+#if defined(_LIBCPP_VERSION)
+extern "C" void __cxa_guard_release(__guard* pg)
+#else
 extern "C" void __cxa_guard_release(__guard* pg) throw()
+#endif
 {
     guard_t* g = reinterpret_cast<guard_t*>(pg);
     const auto scheduler_started = xTaskGetSchedulerState() != taskSCHEDULER_NOT_STARTED;
@@ -189,7 +202,11 @@ extern "C" void __cxa_guard_release(__guard* pg) throw()
     }
 }
 
+#if defined(_LIBCPP_VERSION)
+extern "C" void __cxa_guard_abort(__guard* pg)
+#else
 extern "C" void __cxa_guard_abort(__guard* pg) throw()
+#endif
 {
     guard_t* g = reinterpret_cast<guard_t*>(pg);
     const auto scheduler_started = xTaskGetSchedulerState() != taskSCHEDULER_NOT_STARTED;
diff --git a/components/esp_libc/platform_include/assert.h b/components/esp_libc/platform_include/assert.h
index 1834290fc64..6810b91f3ab 100644
--- a/components/esp_libc/platform_include/assert.h
+++ b/components/esp_libc/platform_include/assert.h
@@ -9,7 +9,7 @@
 */
 #pragma once
 #include <sdkconfig.h>
-#include <stdlib.h>
+//#include <stdlib.h>
 #include <stdint.h>
 
 #include_next <assert.h>
diff --git a/components/esp_system/ld/ld.flash.sections b/components/esp_system/ld/ld.flash.sections
index 15e4d507ed7..9288cb30896 100644
--- a/components/esp_system/ld/ld.flash.sections
+++ b/components/esp_system/ld/ld.flash.sections
@@ -96,6 +96,7 @@
   {
     _flash_rodata_start = ABSOLUTE(.);
 
+    *(.rodata..L.*) 
     SECTION_MAPPINGS(flash_rodata)
 
 #if CONFIG_LIBC_PICOLIBC
diff --git a/components/nvs_flash/src/nvs_page.cpp b/components/nvs_flash/src/nvs_page.cpp
index 2a1226ea850..8f46a3b0d3d 100644
--- a/components/nvs_flash/src/nvs_page.cpp
+++ b/components/nvs_flash/src/nvs_page.cpp
@@ -44,7 +44,7 @@ esp_err_t Page::load(Partition *partition, uint32_t sectorNumber)
         // check if the whole page is really empty
         // reading the whole page takes ~40 times less than erasing it
         const int BLOCK_SIZE = 128;
-        uint32_t* block = new (std::nothrow) uint32_t[BLOCK_SIZE];
+        uint32_t* block = new uint32_t[BLOCK_SIZE];
 
         if (!block) {
             return ESP_ERR_NO_MEM;
diff --git a/tools/cmake/toolchain-clang-esp32c3.cmake b/tools/cmake/toolchain-clang-esp32c3.cmake
index 3b6755d05b7..c517946d9da 100644
--- a/tools/cmake/toolchain-clang-esp32c3.cmake
+++ b/tools/cmake/toolchain-clang-esp32c3.cmake
@@ -19,6 +19,15 @@ set(CMAKE_C_FLAGS "${UNIQ_CMAKE_C_FLAGS}"
     FORCE)
 
 remove_duplicated_flags("--target=riscv32-esp-elf -march=rv32imc_zicsr_zifencei -mabi=ilp32 \
+-fno-aligned-allocation -fno-sized-deallocation -stdlib=libc++ \
+-D_LIBCPP_DISABLE_AVAILABILITY \
+-D_LIBCPP___CONFIG_SITE \
+-D_LIBCPP_ABI_NAMESPACE=__1 \
+-D_LIBCPP_ABI_VERSION=1 \
+-D_LIBCPP_ASSERTION_SEMANTIC_DEFAULT=_LIBCPP_ASSERTION_SEMANTIC_IGNORE \
+-D_LIBCPP_HARDENING_MODE_DEFAULT=_LIBCPP_HARDENING_MODE_NONE \
+-D_LIBCPP_HAS_WIDE_CHARACTERS \
+-D_LIBCPP_VERBOSE_ABORT=\\(void\\) \
                         ${CMAKE_CXX_FLAGS}"
                         UNIQ_CMAKE_CXX_FLAGS)
 set(CMAKE_CXX_FLAGS "${UNIQ_CMAKE_CXX_FLAGS}"
