diff --git a/components/esp_matter/CMakeLists.txt b/components/esp_matter/CMakeLists.txt
index 9a96bc30..c9aad5bf 100644
--- a/components/esp_matter/CMakeLists.txt
+++ b/components/esp_matter/CMakeLists.txt
@@ -110,10 +110,21 @@ endif()
 # This has been added to fix the error and should be removed once fixed:
 # esp-matter/connectedhomeip/connectedhomeip/src/app/EventManagement.cpp:467:23: error: 'writer' is
 # used uninitialized in this function
-idf_build_set_property(COMPILE_OPTIONS "-Wno-error=uninitialized;-Wno-error=maybe-uninitialized;-Wno-missing-field-initializers;" APPEND)
+#idf_build_set_property(COMPILE_OPTIONS "-Wno-error=uninitialized;-Wno-error=maybe-uninitialized;-Wno-missing-field-initializers;" APPEND)
+idf_build_set_property(COMPILE_OPTIONS "-D__FILE__=__FILE_NAME__" APPEND)
+idf_build_set_property(COMPILE_OPTIONS "-Wno-builtin-macro-redefined" APPEND)
+idf_build_set_property(COMPILE_OPTIONS "-Wno-error=uninitialized;-Wno-missing-field-initializers;" APPEND)
+#-Wno-error=maybe-uninitialized;
 idf_build_set_property(COMPILE_OPTIONS "-Wno-error=array-bounds" APPEND)
-
-target_compile_options(${COMPONENT_LIB} PRIVATE "-Wstringop-truncation" "-Wstringop-overflow")
+idf_build_set_property(COMPILE_OPTIONS "-Wno-error=inconsistent-missing-override" APPEND)
+idf_build_set_property(COMPILE_OPTIONS "-Wno-error=sign-conversion" APPEND)
+idf_build_set_property(COMPILE_OPTIONS "-Wno-error=format" APPEND)
+idf_build_set_property(COMPILE_OPTIONS "-Wno-error=missing-braces" APPEND)
+idf_build_set_property(COMPILE_OPTIONS "-Wno-error=#warnings" APPEND)
+idf_build_set_property(COMPILE_OPTIONS "-Wno-error=unused-private-field" APPEND)
+idf_build_set_property(COMPILE_OPTIONS "-Wno-error=non-c-typedef-for-linkage" APPEND)
+
+#target_compile_options(${COMPONENT_LIB} PRIVATE "-Wstringop-truncation" "-Wstringop-overflow")
 
 if (CONFIG_ESP_MATTER_ENABLE_MATTER_SERVER)
     if (CONFIG_ESP_MATTER_ENABLE_DATA_MODEL)
diff --git a/components/esp_matter/data_model/legacy/esp_matter_cluster.cpp b/components/esp_matter/data_model/legacy/esp_matter_cluster.cpp
index 18d0f9bd..fd5fa819 100644
--- a/components/esp_matter/data_model/legacy/esp_matter_cluster.cpp
+++ b/components/esp_matter/data_model/legacy/esp_matter_cluster.cpp
@@ -1352,6 +1352,24 @@ cluster_t *create(endpoint_t *endpoint, config_t *config, uint8_t flags)
             attribute::create_fan_mode_sequence(cluster, config->fan_mode_sequence);
             attribute::create_percent_setting(cluster, config->percent_setting);
             attribute::create_percent_current(cluster, config->percent_current);
+            if (has(feature::multi_speed::get_id())) {
+                feature::multi_speed::add(cluster, &(config->features.multi_speed));
+            }
+            if (has(feature::fan_auto::get_id())) {
+                feature::fan_auto::add(cluster);
+            }
+            if (has(feature::rocking::get_id())) {
+                feature::rocking::add(cluster, &(config->features.rocking));
+            }
+            if (has(feature::wind::get_id())) {
+                feature::wind::add(cluster, &(config->features.wind));
+            }
+            if (has(feature::step::get_id())) {
+                feature::step::add(cluster);
+            }
+            if (has(feature::airflow_direction::get_id())) {
+                feature::airflow_direction::add(cluster, &(config->features.airflow_direction));
+            }
         } else {
             ESP_LOGE(TAG, "Config is NULL. Cannot add some attributes.");
         }
diff --git a/components/esp_matter/data_model/legacy/esp_matter_cluster_impl.h b/components/esp_matter/data_model/legacy/esp_matter_cluster_impl.h
index 5a740c9d..daebc492 100644
--- a/components/esp_matter/data_model/legacy/esp_matter_cluster_impl.h
+++ b/components/esp_matter/data_model/legacy/esp_matter_cluster_impl.h
@@ -315,7 +315,14 @@ typedef struct config {
     nullable<uint8_t> percent_setting;
     uint8_t percent_current;
     void *delegate;
-    config() : fan_mode(0), fan_mode_sequence(2), percent_setting(0), percent_current(0), delegate(nullptr) {}
+    struct {
+        feature::multi_speed::config_t multi_speed;
+        feature::rocking::config_t rocking;
+        feature::wind::config_t wind;
+        feature::airflow_direction::config_t airflow_direction;
+    } features;
+    uint32_t feature_flags;
+    config() : fan_mode(0), fan_mode_sequence(2), percent_setting(0), percent_current(0), delegate(nullptr), feature_flags(0) {}
 } config_t;
 
 cluster_t *create(endpoint_t *endpoint, config_t *config, uint8_t flags);
diff --git a/components/esp_matter_bridge/CMakeLists.txt b/components/esp_matter_bridge/CMakeLists.txt
index 9b7d5eaa..6b3818eb 100644
--- a/components/esp_matter_bridge/CMakeLists.txt
+++ b/components/esp_matter_bridge/CMakeLists.txt
@@ -10,5 +10,5 @@ idf_component_register(SRC_DIRS        ${src_dirs_list}
                        REQUIRES        esp_matter esp_matter_console)
 
 if (CONFIG_ESP_MATTER_ENABLE_DATA_MODEL)
-    target_compile_options(${COMPONENT_LIB} PRIVATE "-Wstringop-truncation" "-Wstringop-overflow")
+#   target_compile_options(${COMPONENT_LIB} PRIVATE "-Wstringop-truncation" "-Wstringop-overflow")
 endif()
diff --git a/components/esp_matter_console/CMakeLists.txt b/components/esp_matter_console/CMakeLists.txt
index ed5cbc26..968a4d6e 100644
--- a/components/esp_matter_console/CMakeLists.txt
+++ b/components/esp_matter_console/CMakeLists.txt
@@ -15,5 +15,5 @@ idf_component_register(SRC_DIRS ${src_dirs}
                     PRIV_REQUIRES ${priv_req})
 
 if (CONFIG_ENABLE_CHIP_SHELL)
-    target_compile_options(${COMPONENT_LIB} PRIVATE "-Wstringop-truncation" "-Wstringop-overflow")
+#   target_compile_options(${COMPONENT_LIB} PRIVATE "-Wstringop-truncation" "-Wstringop-overflow")
 endif()
diff --git a/components/esp_matter_controller/CMakeLists.txt b/components/esp_matter_controller/CMakeLists.txt
index 8f4fb727..07c4424b 100644
--- a/components/esp_matter_controller/CMakeLists.txt
+++ b/components/esp_matter_controller/CMakeLists.txt
@@ -50,5 +50,5 @@ idf_component_register(SRC_DIRS ${src_dirs_list}
 idf_build_set_property(COMPILE_OPTIONS "-Wno-write-strings" APPEND)
 
 if (CONFIG_ESP_MATTER_CONTROLLER_ENABLE)
-    target_compile_options(${COMPONENT_LIB} PRIVATE "-Wstringop-truncation" "-Wstringop-overflow")
+#   target_compile_options(${COMPONENT_LIB} PRIVATE "-Wstringop-truncation" "-Wstringop-overflow")
 endif()
diff --git a/components/esp_matter_ota_provider/CMakeLists.txt b/components/esp_matter_ota_provider/CMakeLists.txt
index 91a7671b..732dcac3 100644
--- a/components/esp_matter_ota_provider/CMakeLists.txt
+++ b/components/esp_matter_ota_provider/CMakeLists.txt
@@ -15,5 +15,5 @@ idf_component_register(SRCS "${srcs}"
                        REQUIRES esp_matter esp_http_client)
 
 if (CONFIG_ESP_MATTER_OTA_PROVIDER_ENABLED)
-    target_compile_options(${COMPONENT_LIB} PRIVATE "-Wstringop-truncation" "-Wstringop-overflow")
+#   target_compile_options(${COMPONENT_LIB} PRIVATE "-Wstringop-truncation" "-Wstringop-overflow")
 endif()
diff --git a/components/esp_matter_rainmaker/CMakeLists.txt b/components/esp_matter_rainmaker/CMakeLists.txt
index 96f70180..a339e723 100644
--- a/components/esp_matter_rainmaker/CMakeLists.txt
+++ b/components/esp_matter_rainmaker/CMakeLists.txt
@@ -12,5 +12,5 @@ idf_component_register(SRCS             ${SRCS_LIST}
                        REQUIRES         ${REQUIRES_LIST})
 
 if (${rainmaker_enabled})
-    target_compile_options(${COMPONENT_LIB} PRIVATE "-Wstringop-truncation" "-Wstringop-overflow")
+#   target_compile_options(${COMPONENT_LIB} PRIVATE "-Wstringop-truncation" "-Wstringop-overflow")
 endif()
diff --git a/connectedhomeip/connectedhomeip b/connectedhomeip/connectedhomeip
--- a/connectedhomeip/connectedhomeip
+++ b/connectedhomeip/connectedhomeip
@@ -1 +1 @@
-Subproject commit 93abd8e6891bb578ea63254fb29d099936f345c8
+Subproject commit 93abd8e6891bb578ea63254fb29d099936f345c8-dirty
diff --git a/device_hal/device/esp32_devkit_c/esp_matter_device.cmake b/device_hal/device/esp32_devkit_c/esp_matter_device.cmake
index a8fb0670..e5560183 100644
--- a/device_hal/device/esp32_devkit_c/esp_matter_device.cmake
+++ b/device_hal/device/esp32_devkit_c/esp_matter_device.cmake
@@ -1,6 +1,6 @@
 cmake_minimum_required(VERSION 3.5)
 if (NOT ("${IDF_TARGET}" STREQUAL "esp32" ))
-    message(FATAL_ERROR "please set esp32 as the IDF_TARGET using 'idf.py set-target esp32'")
+#   message(FATAL_ERROR "please set esp32 as the IDF_TARGET using 'idf.py set-target esp32'")
 endif()
 
 SET(device_type     esp32_devkit_c)
